### PR TITLE
Objects and methods

### PR DESCRIPTION
**Pull request by Azher9767**
objects and methods exercise completed

### CODE REVIEW

File: objects-and-methods/exercise-1/lib/trick_or_treater.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    @costume = costume
+  end
+ 
+  def dressed_up_as
+    @costume.style
+  end
+  # def bag
+  #   @bag = Bag.new
+  # end
+
```
Comment (ahmad19) [inline objects-and-methods/exercise-1/lib/trick_or_treater.rb]: Do we need this ?

File: objects-and-methods/exercise-1/lib/trick_or_treater.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+class TrickOrTreater
+  attr_accessor :bag
+  def initialize(costume)
```
Comment (ahmad19) [inline objects-and-methods/exercise-1/lib/trick_or_treater.rb:4]: Add one line space after line number 2

File: objects-and-methods/exercise-1/lib/trick_or_treater.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    else
+      !@bag.candies.empty?
+    end
+  end
+
+  def candy_count
+    if @bag == nil
+      return 0
+    else
+    @bag.candies.count
```
Comment (ahmad19) [inline objects-and-methods/exercise-1/lib/trick_or_treater.rb:27]: Alignment

File: objects-and-methods/exercise-1/lib/trick_or_treater.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  #   @bag = Bag.new
+  # end
+
+  def has_candy?
+    if @bag == nil
+      return false 
+    else
+      !@bag.candies.empty?
+    end
+  end
```
Comment (ahmad19) [inline objects-and-methods/exercise-1/lib/trick_or_treater.rb]:
```suggestion
  def has_candy?
    return false if @bag.nil?
    
    !@bag.candies.empty?
  end
```

File: objects-and-methods/exercise-1/lib/trick_or_treater.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+  def candy_count
+    if @bag == nil
+      return 0
+    else
+    @bag.candies.count
+    end
+  end
```
Comment (ahmad19) [inline objects-and-methods/exercise-1/lib/trick_or_treater.rb]: Few improvements

1. Do alignment at 27

```suggestion
  def candy_count
    return 0 if @bag == nil

    @bag.candies.count
  end
```

File: objects-and-methods/exercise-2/lib/bag.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    x
+  end
+
+  def take(number)
+    if number == 1
+      return @candies.last
+    else
+      @candies.shift(number)
+    end
+  end
```
Comment (ahmad19) [inline objects-and-methods/exercise-2/lib/bag.rb]:
```suggestion
  def take(number)
    return @candies.last if number == 1

    @candies.shift(number)
  end
```

File: objects-and-methods/exercise-2/lib/trick_or_treater.rb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    @costume.style
+  end
+
+  def has_candy?
+    if @bag == nil
+      return false 
+    else
+      !@bag.candies.empty?
+    end
+  end
```
Comment (ahmad19) [inline objects-and-methods/exercise-2/lib/trick_or_treater.rb]: Add guard clause (one line if condition)

```suggestion
  def has_candy?
    return false if @bag.nil?
    
    !@bag.candies.empty?
  end
```

**Review by ahmad19 — COMMENTED** [review]
added few suggestions

File: objects-and-methods/exercise-1/lib/trick_or_treater.rb (hunk 1 above)
Comment (Azher9767) [reply objects-and-methods/exercise-1/lib/trick_or_treater.rb]: no I will fix it

File: objects-and-methods/exercise-1/lib/trick_or_treater.rb (hunk 2 above)
Comment (Azher9767) [reply objects-and-methods/exercise-1/lib/trick_or_treater.rb:4]: done

**Review by ahmad19 — APPROVED** [review]

**Pull request merged**